### ISSUE TITLE
Traceback from flake8-bugbear

### ISSUE THREAD

**Issue by The-Compiler** [gforcada/flake8-builtins#36]
Similar to #35 I see some weird interaction with flake8-bugbear since upgrading flake8-builtins to 1.4.0. Here's what I get:

```
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/opt/python/3.6.3/lib/python3.6/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/opt/python/3.6.3/lib/python3.6/multiprocessing/pool.py", line 44, in mapstar
    return list(map(*args))
  File "/home/travis/build/qutebrowser/qutebrowser/.tox/flake8/lib/python3.6/site-packages/flake8/checker.py", line 648, in _run_checks
    return checker.run_checks()
  File "/home/travis/build/qutebrowser/qutebrowser/.tox/flake8/lib/python3.6/site-packages/flake8/checker.py", line 579, in run_checks
    self.run_ast_checks()
  File "/home/travis/build/qutebrowser/qutebrowser/.tox/flake8/lib/python3.6/site-packages/flake8/checker.py", line 493, in run_ast_checks
    for (line_number, offset, text, check) in runner:
  File "/home/travis/build/qutebrowser/qutebrowser/.tox/flake8/lib/python3.6/site-packages/bugbear.py", line 37, in run
    visitor.visit(self.tree)
  File "/home/travis/build/qutebrowser/qutebrowser/.tox/flake8/lib/python3.6/site-packages/bugbear.py", line 143, in visit
    super().visit(node)
  File "/opt/python/3.6.3/lib/python3.6/ast.py", line 253, in visit
    return visitor(node)
  File "/opt/python/3.6.3/lib/python3.6/ast.py", line 261, in generic_visit
    self.visit(item)
  File "/home/travis/build/qutebrowser/qutebrowser/.tox/flake8/lib/python3.6/site-packages/bugbear.py", line 143, in visit
    super().visit(node)
  File "/opt/python/3.6.3/lib/python3.6/ast.py", line 253, in visit
    return visitor(node)
  File "/home/travis/build/qutebrowser/qutebrowser/.tox/flake8/lib/python3.6/site-packages/bugbear.py", line 235, in visit_ClassDef
    self.check_for_b903(node)
  File "/home/travis/build/qutebrowser/qutebrowser/.tox/flake8/lib/python3.6/site-packages/bugbear.py", line 414, in check_for_b903
    if len(targets) > 1 or not isinstance(targets[0], ast.Attribute):
IndexError: list index out of range
"""
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/opt/python/3.6.3/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/opt/python/3.6.3/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/travis/build/qutebrowser/qutebrowser/.tox/flake8/lib/python3.6/site-packages/flake8/__main__.py", line 4, in <module>
    cli.main()
  File "/home/travis/build/qutebrowser/qutebrowser/.tox/flake8/lib/python3.6/site-packages/flake8/main/cli.py", line 16, in main
    app.run(argv)
  File "/home/travis/build/qutebrowser/qutebrowser/.tox/flake8/lib/python3.6/site-packages/flake8/main/application.py", line 396, in run
    self._run(argv)
  File "/home/travis/build/qutebrowser/qutebrowser/.tox/flake8/lib/python3.6/site-packages/flake8/main/application.py", line 384, in _run
    self.run_checks()
  File "/home/travis/build/qutebrowser/qutebrowser/.tox/flake8/lib/python3.6/site-packages/flake8/main/application.py", line 310, in run_checks
    self.file_checker_manager.run()
  File "/home/travis/build/qutebrowser/qutebrowser/.tox/flake8/lib/python3.6/site-packages/flake8_per_file_ignores.py", line 33, in run
    orig_run(self)
  File "/home/travis/build/qutebrowser/qutebrowser/.tox/flake8/lib/python3.6/site-packages/flake8/checker.py", line 319, in run
    self.run_parallel()
  File "/home/travis/build/qutebrowser/qutebrowser/.tox/flake8/lib/python3.6/site-packages/flake8/checker.py", line 288, in run_parallel
    for ret in pool_map:
  File "/opt/python/3.6.3/lib/python3.6/multiprocessing/pool.py", line 347, in <genexpr>
    return (item for chunk in result for item in chunk)
  File "/opt/python/3.6.3/lib/python3.6/multiprocessing/pool.py", line 735, in next
    raise value
IndexError: list index out of range
```

**Comment by gforcada** [comment]
@The-Compiler would you mind running it without multiprocessing? i.e. ``flake8 -j1 ...`` TIA

**Comment by The-Compiler** [comment]
Sure!

```
Traceback (most recent call last):
  File "./.tox/flake8/bin/flake8", line 11, in <module>
    sys.exit(main())
  File "/home/florian/proj/qutebrowser/git/.tox/flake8/lib/python3.6/site-packages/flake8/main/cli.py", line 16, in main
    app.run(argv)
  File "/home/florian/proj/qutebrowser/git/.tox/flake8/lib/python3.6/site-packages/flake8/main/application.py", line 396, in run
    self._run(argv)
  File "/home/florian/proj/qutebrowser/git/.tox/flake8/lib/python3.6/site-packages/flake8/main/application.py", line 384, in _run
    self.run_checks()
  File "/home/florian/proj/qutebrowser/git/.tox/flake8/lib/python3.6/site-packages/flake8/main/application.py", line 310, in run_checks
    self.file_checker_manager.run()
  File "/home/florian/proj/qutebrowser/git/.tox/flake8/lib/python3.6/site-packages/flake8_per_file_ignores.py", line 33, in run
    orig_run(self)
  File "/home/florian/proj/qutebrowser/git/.tox/flake8/lib/python3.6/site-packages/flake8/checker.py", line 321, in run
    self.run_serial()
  File "/home/florian/proj/qutebrowser/git/.tox/flake8/lib/python3.6/site-packages/flake8/checker.py", line 305, in run_serial
    checker.run_checks()
  File "/home/florian/proj/qutebrowser/git/.tox/flake8/lib/python3.6/site-packages/flake8/checker.py", line 579, in run_checks
    self.run_ast_checks()
  File "/home/florian/proj/qutebrowser/git/.tox/flake8/lib/python3.6/site-packages/flake8/checker.py", line 493, in run_ast_checks
    for (line_number, offset, text, check) in runner:
  File "/home/florian/proj/qutebrowser/git/.tox/flake8/lib/python3.6/site-packages/bugbear.py", line 37, in run
    visitor.visit(self.tree)
  File "/home/florian/proj/qutebrowser/git/.tox/flake8/lib/python3.6/site-packages/bugbear.py", line 143, in visit
    super().visit(node)
  File "/usr/lib64/python3.6/ast.py", line 253, in visit
    return visitor(node)
  File "/usr/lib64/python3.6/ast.py", line 261, in generic_visit
    self.visit(item)
  File "/home/florian/proj/qutebrowser/git/.tox/flake8/lib/python3.6/site-packages/bugbear.py", line 143, in visit
    super().visit(node)
  File "/usr/lib64/python3.6/ast.py", line 253, in visit
    return visitor(node)
  File "/home/florian/proj/qutebrowser/git/.tox/flake8/lib/python3.6/site-packages/bugbear.py", line 235, in visit_ClassDef
    self.check_for_b903(node)
  File "/home/florian/proj/qutebrowser/git/.tox/flake8/lib/python3.6/site-packages/bugbear.py", line 414, in check_for_b903
    if len(targets) > 1 or not isinstance(targets[0], ast.Attribute):
IndexError: list index out of range
```

**Comment by gforcada** [comment]
@The-Compiler sorry I forgot to ask before, a minimal reproducible code case for that to happen? :sweat_smile:

**Comment by The-Compiler** [comment]
Uh, that's going to be difficult with some 30k line codebase :wink: Can I somehow easily find out what file (and ideally, line) it's happening on?

**Comment by gforcada** [comment]
As the traceback is on bugbear side, if you remove bugbear does it fail? and likewise with only bugbear does it fail? i.e they have both to be present to trigger this bug?

**Comment by gforcada** [comment]
@The-Compiler is the code open so I can have a look myself? :thinking:

**Comment by The-Compiler** [comment]
Sorry, should've been more verbose in my original issue - was a bit in a hurry :smile:

Yeah, the code is at https://github.com/qutebrowser/qutebrowser - at the current master, you can edit `misc/requirements/requirements-flake8.txt` to use `flake8-builtins==1.4.0` and then do `tox -e flake8 -r` to reproduce this.

Indeed when removing flake8-bugbear things work fine as well - maybe someone there (@ambv?) has an idea what's going on?

**Comment by ambv** [comment]
@gforcada, `flake8_builtins.py` is destroying the AST for other plugins to look at on line 116 by calling `stack.pop()` on the targets of an assignment. Copy the list first.

**Comment by dirk-thomas** [comment]
Please see PR #37 which also addresses my problem reported in #34.

**Comment by gforcada** [comment]
1.4.1 is released with #37 on it. Would you mind giving it a try?

Once again, thanks @dirk-thomas for the patch!!
